overlord/devicestate/devicemgr.go

```diff
@@ -462,6 +475,47 @@ func (m *DeviceManager) ensureBootOk() error {
 	return nil
 }
 
+func (m *DeviceManager) ensureInstalled() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	if release.OnClassic {
+		return nil
+	}
+
+	if m.ensureInstalledRan {
+		return nil
+	}
+
+	if m.recoveryMode != "install" {
```

Comment:
we also need to avoid doing registration (become-operation) in install mode

Reply:
Checking recovery mode when becoming operational.